oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/ClaimAccessor.java

```diff
@@ -143,11 +137,15 @@ default URL getClaimAsURL(String claim) {
 	 * @return the claim value or {@code null} if it does not exist or cannot be assigned to a {@code List}
 	 */
 	default List<String> getClaimAsStringList(String claim) {
-		if (!this.containsClaim(claim) || !List.class.isAssignableFrom(this.getClaims().get(claim).getClass())) {
+		if (!containsClaim(claim)) {
 			return null;
 		}
-		List<String> claimValues = new ArrayList<>();
-		((List<?>) this.getClaims().get(claim)).forEach(e -> claimValues.add(e.toString()));
-		return claimValues;
+		Object claimValue = getClaims().get(claim);
+		List<String> convertedValue = ObjectConverters.listStringConverter().convert(claimValue);
```

Comment:
I think Optional is just a bit more readable in situations like these:

```java
return Optional.ofNullable(ObjectConverters.listStringConverter().convert(claimValue))
        .orElseThrow(() -> new IllegalArgumentException...
```

Reply:
I think this is a matter of preference. The current implementation is easily readable as well and it has the benefit of not creating  an extra object (`Optional`) as part of the check.

Reply:
I agree this is a matter of preference. Personally, I find `Optional` harder to read. For me this is especially the case when doing more than one thing on a single line. It is unfortunate, but this often happens because the way IDEs try to format the code with method chaining.

Note that using `Optional` throughout the code base adds quite a bit of object creation which means it can impact GC.